api/src/Service/EventService.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Repository\EventRepository;
+
+class EventService
+{
+
+  /**
+   * @var EventRepository;
+   */
+    protected $eventRepository;
+
+
+    public function __construct(EventRepository $eventRepository)
+    {
+        $this->eventRepository = $eventRepository;
+
+    }
+
+
+    public function getCurrentEvent()
+    {
+        $event = $this->eventRepository->getCurrentEvent();
+        if ($event == null) {
+      // Fallback to last active event in DB
+            $event = $this->eventRepository->getLastActiveEvent();
+        }
+
+        $formatted = [];
+        $formatted["id"] = $event["EventID"];
+        $formatted["name"] = $event["EventName"];
+        $formatted["date_from"] = $event["EventFrom"];
+        $formatted["date_to"] = $event["EventTo"];
+        $formatted["type"] = "event";
+        $formatted["cycle"]["id"] = $event["AnnualCycleID"];
+        $formatted["cycle"]["date_from"] = $event["AnnualCycleFrom"];
+        $formatted["cycle"]["date_to"] = $event["AnnualCycleTo"];
+        $formatted["cycle"]["type"] = "cycle";
+    
+        return $formatted;
+
+    }
```

Comment:
I think we can preserve this value so that we only have to look it up once. This is likely a function that could get called multiple times in a request.

Reply:
I think it depends. Ideally a service using this will call it once. I think you're looking at this from the perspective of includes, right?

Reply:
I am, my brain goes there.  We are going to get to a point where we are likely going to have to look up multiple bits of information from multiple places and so it thought of this.  It is of course not blocking at this point, just being proactive.

Reply:
I think there are probably some ways we can work through that as we get there. I can add the caching for now.